tpm2/transport/tcp/tcp.go

```diff
@@ -0,0 +1,263 @@
+// Package tcp provides access to a TPM over TCP.
+package tcp
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"net"
+)
+
+var (
+	ErrPlatformFailed = errors.New("platform command failed")
+	ErrTPMFailed      = errors.New("TPM command failed")
+	ErrResponseTooBig = errors.New("response too big")
+	ErrTransport      = errors.New("TCP transport error")
+	ErrEmptyResponse  = errors.New("TPM returned empty response (does it need to be powered on?)")
+)
+
+const (
+	maxBufferSize = 1048576
+)
+
+// The de-facto TPM-over-TCP protocol is defined by the Reference Implementation.
+// See https://github.com/TrustedComputingGroup/TPM/blob/main/TPMCmd/Simulator/include/TpmTcpProtocol.h
+
+type regularCommand uint32
```

Comment:
I assume there's no other way to differentiate regular and plat commands other than reading https://github.com/TrustedComputingGroup/TPM/blob/main/TPMCmd/Simulator/src/TcpServer.c?

Reply:
That's right, they are in the same namespace together in the C code.